lib/logger/lib/logger/app.ex

```diff
@@ -77,64 +65,64 @@ defmodule Logger.App do
   Stops the application without sending messages to error logger.
   """
   def stop() do
-    try do
-      Logger.Config.deleted_handlers([])
-    catch
-      :exit, {:noproc, _} ->
-        {:error, {:not_started, :logger}}
-    else
-      deleted_handlers ->
-        result = Application.stop(:logger)
-        add_handlers(deleted_handlers)
-        result
-    end
+    Application.stop(:logger)
   end
 
-  # TODO: Remove conditional error_logger code once we require Erlang/OTP 21+.
-  defp otp_logger? do
-    is_pid(Process.whereis(:logger))
-  end
+  defp add_elixir_handler(otp_reports?, counter) do
+    # TODO: Deprecate this is future releases, probably whole `logger.level`
+    # option and suggest users to use `kernel.logger_level` instead
+    level =
+      case Application.fetch_env!(:logger, :level) do
+        :warn -> :warning
+        other -> other
+      end
```

Comment:
@josevalim we should also think what we should do in case when user explicitly defined the `kernel.logger_level` configuration option as currently we always will lower it to the `debug` (default is `notice`). This can be also a little bit confusing for users that include Elixir libraries to their Erlang apps.